sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py

```diff
@@ -369,28 +380,22 @@ class MetricDefinition(object):  # pylint: disable=too-many-instance-attributes
     """
 
     def __init__(self, **kwargs: Any) -> None:
-        self.dimension_required = kwargs.get(
-            "dimension_required", None
-        )  # type: Optional[bool]
-        self.resource_id = kwargs.get("resource_id", None)  # type: Optional[str]
-        self.namespace = kwargs.get("namespace", None)  # type: Optional[str]
-        self.name = kwargs.get("name", None)  # type: Optional[str]
-        self.unit = kwargs.get("unit", None)  # type: Optional[str]
-        self.primary_aggregation_type = kwargs.get(
-            "primary_aggregation_type", None
-        )  # type: Optional[str]
-        self.supported_aggregation_types = kwargs.get(
-            "supported_aggregation_types", None
-        )  # type: Optional[str]
-        self.metric_availabilities = kwargs.get(
-            "metric_availabilities", None
-        )  # type: List[MetricAvailability]
-        self.id = kwargs.get("id", None)  # type: Optional[str]
-        self.dimensions = kwargs.get("dimensions", None)  # type: Optional[List[str]]
-        self.metric_class = kwargs.get("metric_class", None)  # type: Optional[str]
+        self.dimension_required: Optional[bool] = kwargs.get("dimension_required", None)
+        self.resource_id: Optional[str] = kwargs.get("resource_id", None)
+        self.namespace: Optional[str] = kwargs.get("namespace", None)
+        self.name: Optional[str] = kwargs.get("name", None)
+        self.unit: Optional[Union[str, MetricUnit]] = kwargs.get("unit", None)
+        self.primary_aggregation_type: Optional[Union[str, MetricAggregationType]] = \
+            kwargs.get("primary_aggregation_type", None)
+        self.supported_aggregation_types: Optional[List[Union[str, MetricAggregationType]]] = \
+            kwargs.get("supported_aggregation_types", None)
+        self.metric_availabilities: List[MetricAvailability] = kwargs.get("metric_availabilities", None)
+        self.id: Optional[str] = kwargs.get("id", None)
+        self.dimensions: Optional[List[str]] = kwargs.get("dimensions", None)
+        self.metric_class: Optional[Union[str, MetricClass]] = kwargs.get("metric_class", None)
```

Comment:
This service seems to have a lot of Optional properties - are they all really not sent in the response? This is going to make type checking the samples / how our users write code with the library pretty difficult with a lot of `None` safety checks before accessing the property... if that's how it is, then fine, more feedback for the service than for your PR.

Reply:
Yeeea, all the optionals are kinda a byproduct from how everything is passed in through kwargs. I believe a lot of these are sent in the response when retreiving metrics. I can double check and see If i can make some of these required keyword arguments. and non-optional types.

Reply:
After investigating this, based on the api spec, nothing is explicitly listed as required. And if generating models, every field type is generated as Optional. From testing, seems like all these fields are sent in the response, but I'm not sure if this is guaranteed. I think okay for now to keep these optional.

I do want to eventually use explcit keyword args instead of just kwargs, but not as important since users aren't ever expected to instantiate the classes that have the `**kwargs` init function.

Reply:
Ah got it. Thanks for looking into it.